hudi-common/src/main/java/org/apache/hudi/common/util/BufferedRandomAccessFile.java

```diff
@@ -0,0 +1,411 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.util;
+
+import org.apache.log4j.Logger;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+
+/**
+ * Use a private buffer for the read/write/seek operations of the RandomAccessFile
```

Comment:
@n3nash @nbalajee is this our own implementation?

Reply:
Yes Vinoth.  This is our own implementation of BufferedRandomAccessFile.

Reply:
Thanks! 
@bvaradar @n3nash should we then back out the license change.. I am confused.

Reply:
@vinothchandar @nbalajee : I think we should still retain the LICENSE part and the reference in this file to be conservative as this implementation was based on the Cassandra source code and the structure is essentially same. It is better to err on the side of caution here. Let me know if you strongly think otherwise (cc @n3nash ).